src/bfd-disasm.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <bfd.h>
+#include <dis-asm.h>
+#include "bcc_syms.h"
+#include "bcc_elf.h"
+#include "bfd-disasm.h"
+
+namespace bpftrace {
+
+BfdDisasm::BfdDisasm(std::string &path)
+{
+  fd_ = open(path.c_str(), O_RDONLY);
+}
+
+BfdDisasm::~BfdDisasm()
+{
+  if (fd_ >= 0)
+    close(fd_);
+}
+
+static void get_exec_path(char *tpath, size_t size)
+{
+  const char *path = "/proc/self/exe";
+  ssize_t len;
+
+  len = readlink(path, tpath, size - 1);
+  tpath[len] = 0;
+}
+
+static int fprintf_nop(void *out __attribute__((unused)), const char *fmt __attribute__((unused)), ...)
+{
+  return 0;
+}
+
+static AlignState is_aligned_buf(void *buf, uint64_t size, uint64_t offset)
```

Comment:
Shouldn't we check for `HAVE_UNSAFE_UPROBE` in this function and return NotSupp?

Reply:
hm, we still need to detect if the uprobe is aligned with instructions,
then if there's  `HAVE_UNSAFE_UPROBE` defined, we allow to proceed
with unliagned access by using `--unsafe` option

Reply:
Ah right, my bad